src/Haskell/Ide/Engine/Transport/JsonStdio.hs

```diff
@@ -25,19 +27,27 @@ jsonStdioTransport cin = do
   cout <- newChan :: IO (Chan ChannelResponse)
   hSetBuffering stdout NoBuffering
   let
+    processParse cid r stream = do
+      writeChan cin (wireToChannel cout cid r)
+      rsp <- readChan cout
+      BL.putStr $ A.encode (channelToWire rsp)
+      loop (cid + 1) (P.drop (1::Int) stream)
     loop cid stream = do
       debug "jsonStdioTransport:calling go"
-      (req,stream') <- runStateT decodeMsg stream
+      let splitStream = view (P.break (== fromIntegral (ord '\STX'))) stream
+      (req,res) <- runStateT decodeMsg splitStream
       debug $ T.pack $ "jsonStdioTransport:got:" ++ show req
       case req of
         Just (Left err) -> do
           putStr $ show (HieError (A.String $ T.pack $ show err))
```

Comment:
Good point, I was planning to change the error handling but somehow forgot about it. I'll try to get to it later today.